src/main/java/bio/terra/workspace/service/datareference/DataReferenceService.java

```diff
@@ -78,7 +80,7 @@ public DataReference getDataReferenceByName(
   public DataReference createDataReference(
```

Comment:
For public methods, maybe add to the javadoc that we validate workspace visibility and modifiability first, then validate the data reference operation, etc.